fennel/connectors/connectors.py

```diff
@@ -347,6 +347,14 @@ class Avro(BaseModel):
     token: Optional[str]
 
 
+class Protobuf(BaseModel):
+    registry: str
+    url: str
+    username: Optional[str]
+    password: Optional[str]
+    token: Optional[str]
```

Comment:
Just to confirm, at what level do we do error handling in scenarios:
1. One of username or password is given but not both
2. Token is given along with one or both of username/password. 

It'd be good ensure that doing something like that creates an error with clear message in the mock world itself (ideally within __init__ function itself, not even commit).

Reply:
Assuming this is done (already or after some revisions), could you please add tests to catch these?

Reply:
Added validation and some tests